File: solutions/1758/java/Solution.java
// 1758. Minimum Changes To Make Alternating Binary String
class Solution
{
	public int minOperations(String s)
	{
		int alternatingChangesA = 0;
		int alternatingChangesB = 0;

		for (int i = 0; i < s.length(); ++i)
		{
			char expectedCharA = (char)(i % 2 + '0');

			if (s.charAt(i) != expectedCharA)
			{
				++alternatingChangesA;
			}
			else
			{
				++alternatingChangesB;
			}
		}

		return Math.min(alternatingChangesA, alternatingChangesB);
	}
}

File: solutions/1758/js/solution.js
// 1758. Minimum Changes To Make Alternating Binary String
var minOperations = function(s)
{
	let alternatingChangesA = 0;
	let alternatingChangesB = 0;

	for (let i = 0; i < s.length; ++i)
	{
		let expectedCharA = String.fromCharCode(i % 2 + 48)[0];

		if (s[i] != expectedCharA)
		{
			++alternatingChangesA;
		}
		else
		{
			++alternatingChangesB;
		}
	}

	return Math.min(alternatingChangesA, alternatingChangesB);
};